optimism/src/mips/interpreter.rs

```diff
@@ -699,23 +699,21 @@ pub fn interpret_jtype<Env: InterpreterEnv>(env: &mut Env, instr: JTypeInstructi
         let pos = env.alloc_scratch();
         unsafe { env.bitmask(&instruction, 26, 0, pos) }
     };
+    let instruction_pointer_high_bits = {
+        // FIXME: Requires a range check
+        let pos = env.alloc_scratch();
+        unsafe { env.bitmask(&instruction, 32, 28, pos) }
+    };
+    let target_addr =
+        (instruction_pointer_high_bits * Env::constant(1 << 28)) + (addr * Env::constant(1 << 2));
     match instr {
-        JTypeInstruction::Jump => {
-            // > The address stored in a j instruction is 26 bits of the address
-            // > associated with the specified label. The 26 bits are achieved by
-            // > dropping the high-order 4 bits of the address and the low-order 2
-            // > bits (which would always be 00, since addresses are always
-            // > divisible by 4).
-            // Source: https://max.cs.kzoo.edu/cs230/Resources/MIPS/MachineXL/InstructionFormats.html
-            env.set_instruction_pointer(next_instruction_pointer);
-            env.set_next_instruction_pointer(addr * Env::constant(4));
-            // REMOVEME: when all jtype instructions are implemented.
-            return;
+        JTypeInstruction::Jump => (),
+        JTypeInstruction::JumpAndLink => {
+            env.write_register(&Env::constant(31), instruction_pointer + Env::constant(8));
```

Comment:
There is a constant `REGISTER_RA`.
```suggestion
            env.write_register(&Env::constant(REGISTER_RA), instruction_pointer + Env::constant(8));
```

Reply:
I would like to delete all of those constants.

Reply:
Oh no :(. Ok.